### ISSUE TITLE
GTX 1060 3GB: Not a render node

### ISSUE THREAD

**Issue by ishcabittle** [colinmarc/magic-mirror#37]
I have installed nvidia drivers on ubuntu 24.04, leveraging a gtx 1060 3GB, and when attempting to launch steam via mmserver I get this message:

```
~/mmserver-v0.6.0$ ./mmserver -C steam.toml 
2024-12-13T06:48:29.846629Z  INFO mmserver::vulkan: gpu "NVIDIA GeForce GTX 1060 3GB" ineligible: device "NVIDIA GeForce GTX 1060 3GB" is not a render node
2024-12-13T06:48:29.846797Z  INFO mmserver::vulkan: gpu "llvmpipe (LLVM 17.0.6, 128 bits)" ineligible: device "llvmpipe (LLVM 17.0.6, 128 bits)" is not a render node
Error: no suitable gpu found
```

Is this simply a compatibility issue with this particular card, or do I need to install vulkan libraries other than libvulkan1?  The card does appear in /dev/dri, so I'm unsure where to look next:

```
06:10.0 VGA compatible controller: NVIDIA Corporation GP106 [GeForce GTX 1060 3GB] (rev a1)
06:10.1 Audio device: NVIDIA Corporation GP106 High Definition Audio Controller (rev a1)
```

```
/dev/dri/by-path$ ls -la
total 0
drwxr-xr-x 2 root root  80 Dec 13 06:48 .
drwxr-xr-x 3 root root 100 Dec 13 06:48 ..
lrwxrwxrwx 1 root root   8 Dec 13 06:48 pci-0000:06:10.0-card -> ../card0
lrwxrwxrwx 1 root root  13 Dec 13 06:48 pci-0000:06:10.0-render -> ../renderD128
```

**Comment by ishcabittle** [comment]
/tmp/magic-mirror-003cff25-de41-48d7-af10-32433d080ae6$ cat mmserver.log 
2024-12-13T07:04:59.748952Z DEBUG mmserver: starting up version="mmserver git:8f8d2a2"
2024-12-13T07:04:59.749052Z  WARN mmserver: generating bug report files in: "/tmp/magic-mirror-003cff25-de41-48d7-af10-32433d080ae6"
2024-12-13T07:04:59.750144Z TRACE mmserver::config: using config: Config {
    server: ServerConfig {
        bind: "localhost:9599",
        bind_systemd: false,
        tls_cert: None,
        tls_key: None,
        worker_threads: 8,
        max_connections: Some(
            4,
        ),
        mdns: true,
        mdns_hostname: None,
        mdns_instance_name: None,
    },
    apps: {
        "steam-gamepadui": AppConfig {
            description: None,
            path: [],
            header_image: None,
            command: [
                "steam",
                "-gamepadui",
            ],
            env: {},
            xwayland: true,
            force_1x_scale: false,
            home_isolation_mode: Permanent(
                "/home/ishcabittle/.local/share/mmserver/homes/steam-gamepadui",
            ),
        },
    },
    data_home: "/home/ishcabittle/.local/share/mmserver",
    bug_report_dir: None,
}
2024-12-13T07:04:59.750942Z DEBUG mmserver::vulkan: creating vulkan instance
2024-12-13T07:05:00.176497Z  INFO mmserver::vulkan: gpu "NVIDIA GeForce GTX 1060 3GB" ineligible: device "NVIDIA GeForce GTX 1060 3GB" is not a render node

**Comment by colinmarc** [comment]
Can you post the output of vulkaninfo? Thanks!

**Comment by colinmarc** [comment]
By the way, on ubuntu 24 you'll probably run into this issue: #29 

There are workarounds in that thread.

**Comment by ishcabittle** [comment]
Here is stat renderD128:

```
~/mmserver-v0.6.0$ stat /dev/dri/renderD128
  File: /dev/dri/renderD128
  Size: 0               Blocks: 0          IO Block: 4096   character special file
Device: 0,5     Inode: 746         Links: 1     Device type: 226,128
Access: (0660/crw-rw----)  Uid: (    0/    root)   Gid: (  993/  render)
Access: 2024-12-13 06:48:10.552312067 +0000
Modify: 2024-12-13 06:48:10.552312067 +0000
Change: 2024-12-13 06:48:10.552312067 +0000
 Birth: 2024-12-13 06:48:10.546312068 +0000
```
vulkaninfo's output was long, so I have attached a text file here.
[vulkaninfo.txt](https://github.com/user-attachments/files/18131803/vulkaninfo.txt)

I tried the solution from issue 29 but no dice.  The first two lines are concerning - this Ubuntu 24.04 install does not have a DE, is that a requirement?

**Comment by colinmarc** [comment]
> I tried the solution from issue 29 but no dice.

That's a different issue, which you'll hit once we resolve this one :)

> this Ubuntu 24.04 install does not have a DE, is that a requirement?

Nope, not a requirement.

This is the issue:

```
VkPhysicalDeviceDrmPropertiesEXT:
---------------------------------
        hasPrimary   = false
        hasRender    = false
        primaryMajor = 0
        primaryMinor = 0
        renderMajor  = 0
        renderMinor  = 0
```

Do you have nvidia's weird modeset ko enabled? What's the output of `sudo cat /sys/module/nvidia_drm/parameters/modeset`?

I had to run `update-initramfs -u` on a server to get the modeset kernel module loaded.

**Comment by ishcabittle** [comment]
The output of modeset was:

~/mmserver-v0.6.0$ sudo cat /sys/module/nvidia_drm/parameters/modeset
[sudo] password for ishcabittle: 
N

I ran update-initramfs -u and restarted to the same result.  I'll keep looking into it, having "hasRender = false" as a search term will help a great deal.

**Comment by colinmarc** [comment]
I've had the best result with [this PPA](https://launchpad.net/~graphics-drivers/+archive/ubuntu/ppa) on Ubuntu, maybe you can try that?

Here are the troubleshooting steps I went through a while back for the same issue: https://forums.developer.nvidia.com/t/vk-ext-external-memory-dma-buf-missing-in-545/275834/9

Hope that helps!

**Comment by colinmarc** [comment]
I'm going to close this since it seems to be bad driver install rather than an issue with Magic Mirror - but I'm more than happy to continue helping if you have any more questions, here or in Discord.

**Comment by ishcabittle** [comment]
Burned down the Ubuntu VM, tried on Debian 12, but glibc was too old, and finally settled on Bazzite linux (Fedora 41).  Started right up, so the 1060 3GB is indeed supported, and my driver install was indeed the root cause of this error.

**Comment by colinmarc** [comment]
Is it keeping up with encoding? How is the experience so far?

**Comment by ishcabittle** [comment]
I have the macOS GUI client up, it sees the bazzite hostname and the application I have defined, but when I go to launch I get the MMClientCommon.ClientErrorCancelled(message: "stream closed before request could be received") timeout message.  I figured I should research and see if I could fix it on my own before starting a new issue, I wanted to read through the Discord history and see if this has been solved already.

**Comment by colinmarc** [comment]
> I have the macOS GUI client up, it sees the bazzite hostname and the application I have defined, but when I go to launch I get the MMClientCommon.ClientErrorCancelled(message: "stream closed before request could be received") timeout message. I figured I should research and see if I could fix it on my own before starting a new issue, I wanted to read through the Discord history and see if this has been solved already.

I haven't seen that before! Please upload a bug report to a new issue.
